provider/go.mod

```diff
@@ -0,0 +1,126 @@
+module github.com/pulumi/pulumi-eks/provider/v3
```

Comment:
Curious your thoughts here, we were talking about how the preferred end-goal is one go.mod at the top level of the tree, but bridged template does not yet support it as it insists on the provider/go.mod location. Do you think this is a permanent or temporary addition?

Reply:
This would be a temporary reversal until we update all providers in unison. This scopes the work to onboard ci-mgmt without created additional logic branches.